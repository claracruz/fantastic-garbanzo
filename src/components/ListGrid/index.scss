.list-grid {
    list-style-type: none;
    padding: 0;
}
