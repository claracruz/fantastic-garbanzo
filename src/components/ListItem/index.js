import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';
import {faStarHalfAlt} from '@fortawesome/free-solid-svg-icons'
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

const months = [
	'Jan',
	'Feb',
	'Mar',
	'Apr',
	'May',
	'Jun',
	'Jul',
	'Aug',
	'Sep',
	'Oct',
	'Nov',
	"'Dec"
];


export const ListItem = (props) => {
	const {altmetric_id, details_url, images, published_on, score, title, url} = props.item;
	const datePublished = (published_on) ? new Date(published_on * 1000) : null;

	return (
		<li key={altmetric_id} className="list-item">
			<img src={images.small} alt={title}/>
			<span className="date-published">
				<span className="day">{datePublished && datePublished.getDay()}</span>
		        <span className="month">{months[datePublished && datePublished.getMonth()]}</span>
		        <span className="year">{datePublished && datePublished.getFullYear()}</span>
			</span>
			<div className="content">
				<h3>{title}</h3>

				<div className="details">
					<span className="score">
						<FontAwesomeIcon icon={faStarHalfAlt} />
						<span>{score}</span>
					</span>
					<a href={details_url}
					   className="source"
					   target="_blank"
					   rel="noopener noreferrer" >View metrics data...</a>
					<a href={url}
					   className="source"
					   target="_blank"
					   rel="noopener noreferrer">Go to source...</a>
				</div>
			</div>
		</li>
	);
};


ListItem.propTypes = {
	item: PropTypes.shape({
		altmetric_id: PropTypes.number,
		details_url: PropTypes.string,
		title: PropTypes.string,
		url: PropTypes.string,
		published_on: PropTypes.number,
		score: PropTypes.number,
		images: PropTypes.shape({
			small: PropTypes.string,
			medium: PropTypes.string,
			large: PropTypes.string
		})
	})
};
