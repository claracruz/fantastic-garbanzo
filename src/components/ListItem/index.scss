.list-item {

  background: #ffffff;
  height: auto;
  overflow: hidden;
  width: auto;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.3);
  margin: 15px 0;
  padding: 20px 20px 20px 75px;
  position: relative;
  display: flex;
  text-align: left;

  img {
    flex-shrink: 0;
    width: 64px;
    height: 64px;
  }

  .date-published {
    position: absolute;
    top: 0;
    left: 0;
    background-color: #77d7b9;
    color: #fff;
    padding: 0.8em;

    span {
      display: block;
      text-align: center;
      font-size: 0.8125rem;

      &.day {
        font-weight: 700;
        font-size: 24px;
        text-shadow: 2px 3px 2px rgba(0, 0, 0, 0.18);
      }

      &.month {
        text-transform: uppercase;
      }
    }
  }

  .content {
    margin: 0 25px;

    h3 {
      font-size: 1em;
      width: 75%;
      color: #444;
    }

    a {
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }

    .source {
      color: #0198A6;
      font-size: 0.8125rem;
      font-weight: bold;
      margin-left: 20px;
    }

    .details {
      margin-top: 15px;
      font-size: 0.8125rem;
      font-weight: bold;
      color: #0198A6;

      .score {

        span, a {
          position: relative;
          display: inline-block;
          left: 3px;
          top: 1px;
        }
      }
    }

  }
}
