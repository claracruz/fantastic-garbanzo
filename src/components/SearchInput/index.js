import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import './index.scss';

export class SearchInput extends PureComponent {

	constructor(props) {
		super(props);
		this.state = {
			searchTerm: ''
		}
	}

	onSubmit = () => {
		this.props.onSearch(this.state.searchTerm);
	};

	handleChange = (e) => {
		this.setState({
			searchTerm: e.currentTarget.value
		});
	};

	handleKeyDown = (e) => {
		if (e.key === 'Enter') {
			this.onSubmit();
		}
	};

	render() {
		return (
			<div className="wrap">
				<h2><label htmlFor="search">Search Papers</label></h2>
				<div className="search">
					<input
						id="search"
						className="search-input"
						value={this.state.searchTerm}
						onKeyPress={this.handleChange}
						onKeyDown={this.handleKeyDown}
						onChange={this.handleChange}
						placeholder="Enter a search term"
						type="text" />
					<button className="searchButton" onClick={this.onSubmit}><FontAwesomeIcon icon={faSearch} /></button>
				</div>
			</div>
		);
	}
}

SearchInput.propTypes = {
	onSearch: PropTypes.func.isRequired
};
