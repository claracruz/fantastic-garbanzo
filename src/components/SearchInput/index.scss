
.wrap {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    padding: 20px;
    height: 65px;

    h2 {
        margin-right: 40px;
        position: relative;
        top: 3px;
    }

    .search {
        max-width: 320px;
        width: 100%;
        position: relative;
        margin-right: 35px;
    }

    .search-input {;
        width: 100%;
        border: 3px solid #0198A6;
        padding: 5px;
        height: 36px;
        border-radius: 5px;
        outline: none;
        font-size: 0.8125rem;
        background-color: white;
    }

    .searchButton {
        position: absolute;
        right: -32px;
        width: 40px;
        height: 36px;
        border: 1px solid #0198A6;
        background: #0198A6;
        text-align: center;
        color: #fff;
        border-top-right-radius: 5px;
        border-bottom-right-radius: 5px;
        cursor: pointer;
        font-size: 1rem;
    }
}


