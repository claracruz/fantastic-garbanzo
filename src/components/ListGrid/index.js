import React  from 'react';
import PropTypes from 'prop-types';
import { ListItem } from '../ListItem';
import './index.scss';


function displayData (props) {
	const { items } = props;
	return items.map((item) => {
		return (
			<ListItem key={item.altmetric_id} item={item}/>
		);
	})
}

export const ListGrid = (props) => {
	const { className, items } = props;
	const hasData = items.length > 0;

	return (
		<>
			{
				hasData &&
				<ul className={className}>{displayData(props)}</ul>
			}
		</>
	);
};

ListGrid.propTypes = {
	items: PropTypes.arrayOf(
		PropTypes.shape({
			altmetric_id: PropTypes.number,
			title: PropTypes.string,
			url: PropTypes.string,
			details_url: PropTypes.string,
			published_on: PropTypes.number,
			score: PropTypes.number,
			images: PropTypes.shape({
				small: PropTypes.string,
				medium: PropTypes.string,
				large: PropTypes.string
			}),
		})
	)
};

ListGrid.defaultProps = {
	items: []
};
