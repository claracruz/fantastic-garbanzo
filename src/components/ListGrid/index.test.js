import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { ListGrid } from '../../components/ListGrid';

describe('<ListGrid />', () => {

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<ListGrid />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders correctly', () => {
		const wrapper = shallow(<ListGrid />);
		const instance = wrapper.instance();
		expect(wrapper).toMatchSnapshot();
		expect(instance).toBe(null);

	});
});
