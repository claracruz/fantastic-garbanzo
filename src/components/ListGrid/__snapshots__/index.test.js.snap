// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGrid /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListGrid
    items={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": false,
    },
    "ref": null,
    "rendered": false,
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": false,
      },
      "ref": null,
      "rendered": false,
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
